app/javascript/components/downloads/Downloads.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import axios from 'axios';
+import { PropTypes } from 'prop-types';
+import { Button } from 'react-bootstrap';
+import reportError from '../util/ReportError';
+import { ToastContainer, toast } from 'react-toastify';
+
+class Downloads extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  exportDownloaded = () => {
+    axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
+    axios({
+      method: 'post',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      url: `${window.BASE_PATH}/export/downloaded`,
+      data: { id: this.props.download.id },
+    })
+      .then(() => {
+        toast.success('This download has been marked for removal.', {
+          position: toast.POSITION.TOP_CENTER,
+        });
+      })
+      .catch(err => {
+        reportError(err);
+      });
+  };
+
+  render() {
+    return (
+      <React.Fragment>
+        {this.props.error && (
+          <div>
+            The download link is either invalid, has expired, or the file has already been accessed. Please try exporting again. If you think you received this
+            message in error, please contact the help desk.
```

Comment:
By saying "or the file has already been accessed" would that mean on the backend that the `downloaded` action has been called and the download is now marked for deletion?

Reply:
It would mean that the file has been deleted, not simply marked for deletion.

Reply:
Also when I changed `find_by` to `find_by!` this text really doesn't get hit anymore because `@download` will not be `nil`

Reply:
So with that `find_by!` is the intention to show an error page elsewhere? With a failed `find_by!` by being logged in as a different user I get hit with a `No route matches [GET] "/errors"` page from Rails.

If so, that would mean that the react code for "The download link is either invalid, has expired, or the file has already been accessed. Please try exporting again. If you think you received this message in error, please contact the help desk." would be dead code. 

I think we need to either create an error page that tells the user what's going on when they hit the error in the video I attached, or to change the `find_by!` to enable the user to be presented with the language about what went wrong.

https://user-images.githubusercontent.com/24628687/113765576-b1345a80-96d9-11eb-90e2-3babd1fa515d.mov

Reply:
You're presented with the generic 404 page. You need to change `consider_local_requests = true` to false to show the proper error page in development environment. I guess the question is do we provide "more security" via less information and serve our generic 404, or do we provide more information and use a separate nil check and return the error text

Reply:
Okay, I think a generic 404 would be acceptable. BUT if you take that route then the component needs to be updated.

Reply:
Please reexamine all of the authentication behavior since I've removed downloads controller and pulled all features into proxy controller